<launch>
 <!-- Minitaur State Machine Node-->
  <node name="spot_sm" pkg="mini_ros2" exec="spot_sm" output="screen">
    <param name="frequency" value="200.0" />
  </node>

 <!-- Joystick Node -->
  <node respawn="true" pkg="joy"
        exec="joy_node" name="spot_joy" >
    <param name="dev" value="/dev/input/js0" />
    <param name="deadzone" value="0.005" />
  </node>

 <!-- Teleop Node -->
 <node name="spot_teleop" pkg="mini_ros2" exec="teleop_node" output="screen">
  <param name="frequency" value="200.0"/>
  <param name="axis_linear_x" value="2"/>
  <param name="axis_linear_y" value="3"/>
  <param name="axis_linear_z" value="1"/>
  <param name="axis_angular" value="0"/>
  <param name="scale_linear" value="1.0"/>
  <param name="scale_angular" value="1.0"/>
  <param name="button_switch" value="0"/>
  <param name="button_estop" value="1"/>
</node>

<!-- Spot Controller -->
<node name="spot_real" pkg="mini_ros2" exec="spot_real_interface" output="screen">
<!-- Agent Number for ARS (GMBC) Policy. Default (0) Loads no Agent. Policy numbers start at 9 and increase by 10. E.G: 9...19...29..."-->
<param name="agent_num" value="9" />
<param from="$(find-pkg-prefix mini_ros2)/config/spot_params.yaml" />
<param from="$(find-pkg-prefix mini_ros2)/config/joy_params.yaml" />
<param from="$(find-pkg-prefix mini_ros2)/config/policy_params.yaml" />
</node>

<!-- Sensor Interface Node -->
<!-- TODO: replace with micro-ROS or similar?
<node name="serial_node" pkg="rosserial_python" exec="serial_node.py">
    <param name="port" value="/dev/ttyS0"/>
    <param name="baud" value="500000"/>
</node>
-->

<!-- NOTE:
        Setting Up Joystick:
          - Get Number (you will see something like jsX): ls /dev/input/
          - Make available to ROS: sudo chmod a+rw /dev/input/jsX
          - Make sure <param name="dev" value="/dev/input/jsX" /> is correct in launch

        You can ignore this msg: [ERROR] [1591631380.406690714]: Couldn't open joystick force feedback!
        It just means your controller is missing some functionality, but this package doesn't use it. 
-->

</launch>